File: Backend/htbp_parent/service/service_hosp/src/main/java/com/seclib/htbp/hosp/repository/HospitalRepository.java
package com.seclib.htbp.hosp.repository;

import com.seclib.htbp.model.hosp.Hospital;
import org.springframework.data.mongodb.repository.MongoRepository;

import java.util.List;

public interface HospitalRepository extends MongoRepository<Hospital,String> {
    Hospital getHospitalByHoscode(String hoscode);

    List<Hospital> findHospitalByHosnameLike(String hosname);
}

File: Backend/htbp_parent/service/service_hosp/src/main/java/com/seclib/htbp/hosp/repository/HospitalRepository.kt
package com.seclib.htbp.hosp.repository

import org.springframework.data.mongodb.repository.MongoRepository
import com.seclib.htbp.model.hosp.Hospital

interface HospitalRepository : MongoRepository<Hospital?, String?> {
    fun getHospitalByHoscode(hoscode: String?): Hospital?
    fun findHospitalByHosnameLike(hosname: String?): List<Hospital?>?
}